lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp

```diff
@@ -1857,3 +1895,372 @@ sgw_eps_bearer_context_information_t* update_sgw_context_to_s11_teid_map(
+    OAILOG_ERROR_UE(module, sgw_ctxt_p->imsi64,
+                    "Failed to get default bearer context\n");
+    OAILOG_FUNC_RETURN(module, RETURNerror);
+  }
+
+  eps_bearer_ctxt_p->eps_bearer_id = 0;
+  eps_bearer_ctxt_p->paa = default_eps_bearer_entry_p->paa;
+  // SGW FTEID
+  eps_bearer_ctxt_p->s_gw_teid_S1u_S12_S4_up = s1_u_sgw_fteid;
+
+  if (pdn_type == IPv4 || pdn_type == IPv4_AND_v6) {
+    eps_bearer_ctxt_p->s_gw_ip_address_S1u_S12_S4_up.pdn_type = IPv4;
+    eps_bearer_ctxt_p->s_gw_ip_address_S1u_S12_S4_up.address.ipv4_address
+        .s_addr = sgw_ip_address_S1u_S12_S4_up;
+  } else {
+    eps_bearer_ctxt_p->s_gw_ip_address_S1u_S12_S4_up.pdn_type = IPv6;
+    memcpy(
+        &eps_bearer_ctxt_p->s_gw_ip_address_S1u_S12_S4_up.address.ipv6_address,
+        &sgw_ipv6_address_S1u_S12_S4_up,
+        sizeof(eps_bearer_ctxt_p->s_gw_ip_address_S1u_S12_S4_up.address
+                   .ipv6_address));
+  }
+  // DL TFT
+  memcpy(&eps_bearer_ctxt_p->tft, &bearer_req_p->dl_tft,
+         sizeof(traffic_flow_template_t));
+  // QoS
+  memcpy(&eps_bearer_ctxt_p->eps_bearer_qos, &bearer_req_p->eps_bearer_qos,
+         sizeof(bearer_qos_t));
+  // Save Policy Rule Name
+  snprintf(eps_bearer_ctxt_p->policy_rule_name, POLICY_RULE_NAME_MAXLEN + 1,
+           "%s", bearer_req_p->policy_rule_name);
+  eps_bearer_ctxt_p->sgw_sequence_number = sequence_number;
+  OAILOG_INFO_UE(module, sgw_ctxt_p->imsi64,
+                 "Number of DL packet filter rules: %d\n",
+                 eps_bearer_ctxt_p->tft.numberofpacketfilters);
+
+  // Create temporary spgw bearer context entry
+  pgw_ni_cbr_proc_t* pgw_ni_cbr_proc =
+      pgw_get_procedure_create_bearer(sgw_ctxt_p);
+  if (!pgw_ni_cbr_proc) {
+    OAILOG_DEBUG_UE(module, sgw_ctxt_p->imsi64,
+                    "Creating a new temporary eps bearer context entry\n");
+    pgw_ni_cbr_proc = pgw_create_procedure_create_bearer(sgw_ctxt_p);
+    if (!pgw_ni_cbr_proc) {
+      OAILOG_ERROR_UE(module, sgw_ctxt_p->imsi64,
+                      "Failed to create temporary eps bearer context entry\n");
+      OAILOG_FUNC_RETURN(module, RETURNerror);
+    }
+  }
+  struct sgw_eps_bearer_entry_wrapper_s* sgw_eps_bearer_entry_p =
+      new sgw_eps_bearer_entry_wrapper_s();
+  sgw_eps_bearer_entry_p->sgw_eps_bearer_entry = eps_bearer_ctxt_p;
+  LIST_INSERT_HEAD((pgw_ni_cbr_proc->pending_eps_bearers),
+                   sgw_eps_bearer_entry_p, entries);
+  OAILOG_FUNC_RETURN(module, RETURNok);
+}
+
+// Build and send ITTI message, s11_create_bearer_request to MME APP
+status_code_e sgw_build_and_send_s11_create_bearer_request(
+    sgw_eps_bearer_context_information_t* sgw_eps_bearer_context_information,
+    const itti_gx_nw_init_actv_bearer_request_t* const bearer_req_p,
+    pdn_type_t pdn_type, uint32_t sgw_ip_address_S1u_S12_S4_up,
+    struct in6_addr* sgw_ipv6_address_S1u_S12_S4_up, teid_t s1_u_sgw_fteid,
+    log_proto_t module) {
+  OAILOG_FUNC_IN(module);
+  MessageDef* message_p = NULL;
```

Comment:
NULL->nullptr

Reply:
MessageDef is part of itti structure which is in C, on failure it would return NULL